app/Exceptions/Handler.php

```diff
@@ -108,9 +113,24 @@ public function render($request, Exception $e)
                 }
             }
 
-            return response()->json([
+            $response = response()->json([
                 'errors' => $errors
             ], $statusCode, $headers);
+
+            // In the circumstance where an exception is raised
+            // before the lumen request cycle reaches the middleware stage,
+            // it is necessary to force the inclusion of the CORS headers.
+
+            // This uses the app's CORS config.
+            // This config must be loaded in the bootstrap process
+            // before exception handlers are expected to be used.
+            $options = config('cors');
+
+            // This CorsService relies on Asm89\Stack
+            $cors = new CorsService($options);
+            $response = $cors->addActualRequestHeaders($response, $request);
+
+            return $response;
         }
 
         return parent::render($request, $e);
```

Comment:
Since we're not changing this one, the change will only take effect for the json requests...we might want to fix this for other requests but it's not crucial.